internal/runbits/dependencies/changesummary.go

```diff
@@ -41,6 +41,27 @@ func OutputChangeSummary(out output.Outputer, newBuildPlan *buildplan.BuildPlan,
 		}
 	}
 
+	if len(changeset.Added) > 0 && len(addedLocale) == 0 {
+		for _, a := range changeset.Added {
+			// Check if the added artifact is a direct dependency of a requested package.
+			// This can happen when an updated package brings in a new dependency.
+			for _, req := range requested {
+				if !isDirectDependency(a, req) {
+					continue
+				}
+				v := fmt.Sprintf("%s@%s", req.Name(), req.Version()) // requested package, not added package
+				addedString = append(addedLocale, v)
+				addedLocale = append(addedLocale, fmt.Sprintf("[ACTIONABLE]%s[/RESET]", v))
+
+				for _, i := range a.Ingredients { // added package, not requested package
+					dependencies = append(dependencies, i)
+					directDependencies = append(dependencies, i)
+				}
+				break
+			}
+		}
+	}
```

Comment:
While this might get us the result we want, it feels too implicit. Is there a reason we can't iterate over `changeset.Updated` the same way we do over `changeset.Added`? We'll still need to calculate dependencies that were actually changed, but at least that way we are explicitly interpreting just the requirements that were updated.

I'm worried the current solution will break in the future or cause edge case bugs.

Reply:
Okay, I've switched to iterating over `changeset.Upated` and then looking for a direct dependency in `changeset.Added`. Let me know if this isn't what you meant.